### PR TITLE
orc tests: add more cases

### PR DESCRIPTION
**Pull request by deepthi**
Partial fix for #6769 

Signed-off-by: deepthi <deepthi@planetscale.com>

### CODE REVIEW

File: go/test/endtoend/orchestrator/orc_test.go [hunk 1]
```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
...
+	assert.NotNil(t, curMaster, "should have elected a master")
 
-		// Start the tablet
-		err := tablet.VttabletProcess.Setup()
-		require.NoError(t, err)
+	// Make the current master database unavailable.
+	err := curMaster.MysqlctlProcess.Stop()
+	require.NoError(t, err)
+
+	for _, tablet := range shard0.Vttablets {
```
Comment (shlomi-noach) [inline go/test/endtoend/orchestrator/orc_test.go:159]: where's the `Sleep()` now?

File: go/test/endtoend/orchestrator/orc_test.go [hunk 2]
```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
...
+	// Make the current master database read-only.
+	runSQL(t, "set global read_only=ON", curMaster)
 
+	// wait for repair
+	// TODO(deepthi): wait for condition instead of sleep
+	time.Sleep(15 * time.Second)
+	qr := runSQL(t, "select @@global.read_only", curMaster)
+	require.NotNil(t, qr)
+	require.Equal(t, 1, len(qr.Rows))
+	require.Equal(t, "[[INT64(0)]]", fmt.Sprintf("%s", qr.Rows), qr.Rows)
```
Comment (shlomi-noach) [inline go/test/endtoend/orchestrator/orc_test.go:197]: :+1:

File: go/test/endtoend/orchestrator/orc_test.go [hunk 3]
```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
...
+	runSQL(t, "set global read_only=OFF", replica)
+
+	// wait for repair
+	// TODO(deepthi): wait for condition instead of sleep
+	time.Sleep(15 * time.Second)
+	qr := runSQL(t, "select @@global.read_only", replica)
+	require.NotNil(t, qr)
+	require.Equal(t, 1, len(qr.Rows))
+	require.Equal(t, "[[INT64(1)]]", fmt.Sprintf("%s", qr.Rows), qr.Rows)
+}
```
Comment (shlomi-noach) [inline go/test/endtoend/orchestrator/orc_test.go:238]: :+1:

File: go/test/endtoend/orchestrator/orc_test.go [hunk 4]
```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
...
 	}
+	require.NotNil(t, replica, "should be able to find a replica")
+	// use vtctlclient to stop replication
+	_, err = clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("StopReplication", replica.Alias)
+	require.NoError(t, err)
+
+	// wait for repair
+	time.Sleep(15 * time.Second)
+	// check replication is setup correctly
+	checkReplication(t, clusterInstance, curMaster, []*cluster.Vttablet{replica})
```
Comment (shlomi-noach) [inline go/test/endtoend/orchestrator/orc_test.go:276]: I feel like these tests share a lot of code in common, the code should probably be refactored to provide a framework for setup-destroy-wait-expect cycle.

**Review by shlomi-noach — APPROVED** [review]
looks good, some improvements suggested, not necessarily for this PR

File: go/test/endtoend/orchestrator/orc_test.go (hunk 1 above)
Comment (deepthi) [reply go/test/endtoend/orchestrator/orc_test.go:159]: It is inside the call to checkMasterTablet

File: go/test/endtoend/orchestrator/orc_test.go (hunk 4 above)
Comment (deepthi) [reply go/test/endtoend/orchestrator/orc_test.go:276]: I moved as much as possible into createCluster and checkReplication. What is left is slightly different in each test case.
As we add more tests we can revisit and refactor.

**Pull request merged**